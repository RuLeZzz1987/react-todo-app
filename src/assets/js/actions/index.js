export CategoryActions from './CategoryActions';
export TodoActions from './TodoActions';
export FilterActions from './FilterActions';