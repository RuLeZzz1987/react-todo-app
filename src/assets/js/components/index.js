import React, { Component } from 'react';
import Header from './header';
import Main from './main';
import { Category, Todo } from '../models';

class App extends Component {
    
    constructor(props) {
        super(props);
        
        this.state = {
            categories: [new Category({name: 'Category_1', isRoot: true, children: [new Todo({name: 'To-Do Item #1'}), new Category({name: 'Category_1_1'})]})]
        };
        
        this.updateCategories = (categories, cb) => this.setState({categories}, cb);
       
    }
    
    render() {
        return (
            <div>
                <Header
                    categories={this.state.categories}
                />
                <Main
                    categories={this.state.categories}
                    updateCategories={this.updateCategories}
                />
            </div>
        )
    }
}

export default App;