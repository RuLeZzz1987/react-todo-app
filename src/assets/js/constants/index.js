
export const CATEGORY = Symbol('category');

export const TODO = Symbol('todo');