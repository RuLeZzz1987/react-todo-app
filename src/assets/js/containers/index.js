export Header from './HeaderContainer';
export Categories from './CategoriesContainer';
export Category from './CategoryContainer';
export Editor from './EditorContainer';
export ModalError from './ModalErrorContainer';