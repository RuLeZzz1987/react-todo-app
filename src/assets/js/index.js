import React from 'react';
import { render } from 'react-dom';
import App from './components'
import '../styles/styles.scss';

render(<App/>, document.querySelector('#root'));