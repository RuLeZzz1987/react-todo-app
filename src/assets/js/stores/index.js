export TodoStore from './TodoStore';
export CategoryStore from './CategoryStore';
export FilterStore from './FilterStore';
export ErrorStore from './ErrorStore';