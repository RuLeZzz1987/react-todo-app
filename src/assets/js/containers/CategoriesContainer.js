import React, { Component } from 'react';
import { Container } from 'flux/utils'
import { CategoryStore, FilterStore, TodoStore } from '../stores';
import Categories from '../components/categories';
import { CategoryActions } from '../actions';
import { reduceWithCondition } from '../helpers';

class CategoriesContainer extends Component {

  static getStores() {
    return [CategoryStore, FilterStore, TodoStore]
  }

  static calculateState(prevState, props) {
    const { categoryId, todoId } = props.params;
    const todos = TodoStore.getState();
    const filters = FilterStore.getState();
    let categories = reduceWithCondition(CategoryStore.getState())('parentId', undefined);

    if (!filters.showDone) {
      categories = reduceWithCondition(categories)('isComplete', false)
    }

    return {
      ids: Object.keys(categories),
      selectedId: props.categoryId,
      isTodoFound: Object.keys(todos).some(id => id == todoId && todos[id].parentId == categoryId)
    }
  }

  render() {
    return (
      <Categories
        {...this.state}
        {...this.props}
        addCategory={CategoryActions.addCategory}
      />
    )
  }
}

export default Container.create(CategoriesContainer, {withProps: true})