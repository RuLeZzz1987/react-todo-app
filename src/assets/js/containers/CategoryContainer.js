import React, { Component } from 'react';
import { Container } from 'flux/utils';
import { CategoryStore } from '../stores';
import { Category } from '../components/common';
import { CategoryActions } from '../actions';

class CategoryContainer extends Component {

  static getStores() {
    return [CategoryStore]
  }

  static calculateState(prevState, props) {
    const categories = CategoryStore.getState();
    const category = categories[props.id];

    return {
      categories,
      category,
    }
  }

  render() {
    if (!this.state.category) return null;

    return (
      <Category
        {...this.state}
        {...this.props}
        addCategory={CategoryActions.addCategory}
        editCategory={CategoryActions.editCategory}
        removeCategory={CategoryActions.removeCategory}
        moveTo={CategoryActions.moveTodoToOther}
      />
    )
  }

}

export default Container.create(CategoryContainer, {withProps: true})