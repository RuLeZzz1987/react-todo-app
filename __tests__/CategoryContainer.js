import React from 'react';
import renderer from 'react-test-renderer';
import { Category } from '../src/assets/js/containers';
import * as Stores from '../src/assets/js/stores';

describe('CategoryContainer', function () {

  beforeEach(function () {
    Object.keys(Stores).forEach(key=>{
      Stores[key]._state = Stores[key].getInitialState();
    });

    this.render = ({params={}, id, isTodoFound=false}) => renderer.create(
      <Category
        id={id}
        params={params}
        isTodoFound={isTodoFound}
      />
    ).toJSON();

  });

  it('should return null on category missing', function () {

    expect(this.render('5')).toMatchSnapshot();

  });

});
