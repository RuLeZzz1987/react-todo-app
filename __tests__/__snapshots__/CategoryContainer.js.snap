exports[`CategoryContainer should return null on category missing 1`] = `null`;
